mac/Keyman4MacIM/Keyman4MacIM/KeySender.m

```diff
@@ -90,23 +63,20 @@ - (void)postKeyboardEventWithSource: (CGEventSourceRef)source code:(CGKeyCode) v
 - (void)sendKeymanKeyCodeForEvent:(NSEvent *)event {
   [self.appDelegate logDebugMessage:@"KeySender sendKeymanKeyCodeForEvent"];
   
-  [self sendKeyDown:kKeymanEventKeyCode forSourceEvent:event includeKeyUp:NO];
-  
+  ProcessSerialNumber psn;
+
   // Returns the frontmost app, which is the app that receives key events.
   NSRunningApplication *app = NSWorkspace.sharedWorkspace.frontmostApplication;
   pid_t processId = app.processIdentifier;
   NSString *bundleId = app.bundleIdentifier;
+  GetProcessForPID(processId, &psn);
   
   [self.appDelegate logDebugMessage:@"sendKeymanKeyCodeForEvent keyCode %lu to app %@ with pid %d", (unsigned long)kKeymanEventKeyCode, bundleId, processId];
   
-  CGEventRef cgevent = [event CGEvent];
-  
-  // use source from event to generate new event
-  CGEventSourceRef source = CGEventCreateSourceFromEvent(cgevent);
-  CGEventRef keyDownEvent = CGEventCreateKeyboardEvent(source, kKeymanEventKeyCode, true);
-  
-  // TODO: add version check
-  CGEventPostToPid(processId, keyDownEvent);
+  // use nil as source, as this generated event is not directly tied to the originating event
+  CGEventRef keyDownEvent = CGEventCreateKeyboardEvent(nil, kKeymanEventKeyCode, true);
```

Comment:
This is a change in behaviour. Are there possible ramifications?

Reply:
This is a change from my previous version of the code, but Keyman 16 specifies nil as the source for this event, so I decided to switch back to that as I could see no difference in the behavior. The impact of the event source is not clearly documented. It seems reasonable that if any event we generate is going to have a nil event source it would be this one because it is a message to ourselves, and we do not want it to be interpreted by anyone but Keyman.